src/machine.rs

```diff
@@ -249,6 +249,37 @@ impl<'m> Machine<'m> {
         Ok(())
     }
 
+    /// Delete the machine.
+    ///
+    /// Deletes the machine, cleaning up all associated resources.
+    #[instrument(skip_all)]
+    pub async fn delete(self) -> Result<(), Error> {
+        let vm_id = self.config.vm_id();
+
+        if self.is_running() {
+            // Trying to shut down gracefully
+            self.shutdown().await?;
+            info!("{vm_id}: Waiting for the jailer to shut down...");
+            sleep(Duration::from_secs(10)).await;
+        }
+
+        if self.is_running() {
```

Comment:
we don't need to check at all. Just force shutdown and don't propagate errors from it (but `trace!` them).